extensions/gadgets/base.py

```diff
@@ -84,7 +84,8 @@ def html_body(self):
         once the necessary attributes are supplied.
         """
         html_templates = utils.get_file_contents(os.path.join(
-            feconf.GADGETS_DIR, self.type, '%s.html' % self.type))
+            utils.get_asset_dir_prefix()[1:], feconf.GADGETS_DIR,
```

Comment:
Yeah, I worry a bit about this, especially since this is causing changes to app.yaml as well. Do you think we could do the file hashes for everything except the gadgets/etc. in this PR, and leave the processing of extensions-html stuff for later until #2575 is handled?